### PR TITLE
dev-vcs/gitstrap: added new package "dev-vcs/gitstrap-0.1.7"

### PR DESCRIPTION
**Pull request by g4s8**
Added new "gitstrap" package - it's command line tool which helps to
automate routine Github repository operations.

Bug: https://bugs.gentoo.org/676864
Closes: https://bugs.gentoo.org/676864
Package-Manager: Portage-2.3.51, Repoman-2.3.11

### CODE REVIEW

**Comment by g4s8** [comment]
@jer-gentoo could you please look at this PR? You started review in https://bugs.gentoo.org/676864
I fixed issues in ed306740d11b5c5b33ac2f2e1b30fd991509cfac

**Comment by g4s8** [comment]
@jer-gentoo ping, please look at https://github.com/gentoo/gentoo/pull/10936#issuecomment-462424618

**Comment by juippis** [comment]
Jer doesn't really look at github submissions. 

You're missing a sign-off that's required to get this merged, check the that gentoo-bot gave about it. You need to accept the terms.

File: dev-util/gitstrap/gitstrap-0.1.7.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	emake OUTPUT=lib BUILD_VERSION=${PV} BUILD_DATE=${now} lib || die
+	go build -o ${PN} ./cmd/${PN} || die
+}
+
+src_test() {
+	emake OUTPUT=lib test || die
+}
+
+src_install() {
+	dobin ${PN} || die "${PN} installation failed"
```
Comment (juippis) [inline dev-util/gitstrap/gitstrap-0.1.7.ebuild]: This doesn't need `die`, also the emakes above don't need it, but you could add a clear message to see where it dies, if you want to keep them.

File: dev-util/gitstrap/gitstrap-0.1.7.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+DESCRIPTION="Command line tool to bootstrap Github repository"
+HOMEPAGE="https://github.com/g4s8/${PN}"
+SRC_URI="${HOMEPAGE}/archive/${PV}.tar.gz -> ${PN}-${PV}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+DEPEND="dev-lang/go"
```
Comment (juippis) [inline dev-util/gitstrap/gitstrap-0.1.7.ebuild]: It just hit me, isn't there some sort of eclass that makes packaging go packages a lot easier? Making whatever you do below redundant? (To be honest this ebuild looks a bit weird, but I've got no knowledge about go packages)

**Comment by g4s8** [comment]
@juippis thanks, I added signoff with second commit https://github.com/gentoo/gentoo/pull/10936/commits/cc64b54db525a7f48763caf5ea3dc48a340afa5c is it ok, or I should close this PR and submit new one with signoff in first commit?

**Comment by juippis** [comment]
Well you should squash your 3 commits into one, with added sign-off.

**Comment by g4s8** [comment]
@juippis thanks for review. I updated ebuild where you asked and found `golang-base` eclass which can be used here (it's not possible to use `golang-build` because it uses another layout directory with single `main.go` file in root instead of `/cmd` [directory layout](https://github.com/golang-standards/project-layout#go-directories))

**Comment by g4s8** [comment]
Actually I found a way how to hack `golang-build` eclass to work with my code: I just overrode `get_golibdir_gopath` and replaced `src_install` with custom function, and now it's working fine. @juippis thanks for suggestion.

**Pull request closed without merging**